ext/hal/nordic/nrfx_glue.h

```diff
@@ -233,16 +233,39 @@ extern "C" {
 //------------------------------------------------------------------------------
 
 /** @brief Bitmask that defines DPPI channels that are reserved for use outside of the nrfx library. */
-#define NRFX_DPPI_CHANNELS_USED  0
+#define NRFX_DPPI_CHANNELS_USED 0
 
 /** @brief Bitmask that defines DPPI groups that are reserved for use outside of the nrfx library. */
-#define NRFX_DPPI_GROUPS_USED    0
+#define NRFX_DPPI_GROUPS_USED   0
 
 /** @brief Bitmask that defines PPI channels that are reserved for use outside of the nrfx library. */
-#define NRFX_PPI_CHANNELS_USED  0
+#define NRFX_PPI_CHANNELS_USED  (NRFX_PPI_CHANNELS_USED_BY_BT_CTLR | \
+                                 NRFX_PPI_CHANNELS_USED_BY_PWM_SW)
+
+#if IS_ENABLED(CONFIG_BT_CTLR)
+extern const u32_t z_bt_ctlr_used_nrf_ppi_channels;
+#define NRFX_PPI_CHANNELS_USED_BY_BT_CTLR   z_bt_ctlr_used_nrf_ppi_channels
+#else
+#define NRFX_PPI_CHANNELS_USED_BY_BT_CTLR   0
+#endif
+
+#if IS_ENABLED(CONFIG_PWM_NRF5_SW)
+#define NRFX_PPI_CHANNELS_USED_BY_PWM_SW \
```

Comment:
@anangl does this apply for nRF91, as well?

Reply:
Currently you can't compile the pwm_nrf5_sw driver for nRF91.

Reply:
Yes, so i wonder: when we will have PWM for 91, you would need to add the _DPPI_ macros in this ifdef block.
Alternatively,

Reply:
> Yes, so i wonder: when we will have PWM for 91, you would need to add the _DPPI_ macros in this ifdef block.

Yes, after the pwm_nrf5_sw driver is improved to support nRF91, I will update this part of `nrfx_glue.h`. Actually, I planned to do such update directly in #15287 once this PR is merged.